gson/src/test/java/com/google/gson/stream/JsonReaderTest.java

```diff
@@ -642,6 +642,7 @@ private void assertNotANumber(String s) throws IOException {
       strictReader.nextDouble();
       fail("Should have failed reading " + s + " as double");
     } catch (MalformedJsonException e) {
+      // OK:  Should fail because `s` can't be read as double
```

Comment:
If you want you can instead perform an assertion on `e` to make sure it has the expected message.